deploy/terraform/terraform-units/modules/sap_system/hdb_node/variables_local.tf

```
@@ -241,4 +252,18 @@ locals {
       }
     ]
   ])
+
+  disk_count = length(local.data-disk-list) / length(local.hdb_vms)
```

it can be simpler:
```suggestion
  disk_count =  length(local.data-disk-per-dbnode)
```
Another option would be to have the lun number in `local.data-disk-list`, so don't need to calculate later in the resource block.

---

I was thinking of that but did not know how to do that as there is a nested loop

---

For hdb is it very straight forward since it has two nested loops:

```
data-disk-list = flatten([
    for hdb_vm in local.hdb_vms : [
      for idx, datadisk in local.data-disk-per-dbnode : {
        vm_name              = hdb_vm.name
        disk_name            = datadisk.disk_name
        storage_account_type = datadisk.storage_account_type
        lun                  = idx
      }
    ]
  ])
```

I can take a look at anydb case.